components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/OptimizedApplicationConfig.java

```diff
@@ -61,7 +63,7 @@ private class OptimizedAuthStep implements Serializable {
 
         private int stepOrder;
         private List<String> localAuthenticatorConfigNames;
-        private List<String> federatedIdPResourceIds;
+        private List<OptimizedFederatedIdP> optimizedFederatedIdPs;
```

Comment:
can we keep the `federatedIdPResourceIds`, so that the existing data will also work and we don't need to update the serialVersionUID.

In the new implementation, we can both cases (federatedIdPResourceIds is empty - new data, federatedIdPResourceIds not empty - old data)

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/commit/87aac03e1fe4a099ff0d87cfced2667a1a3f74ad & https://github.com/wso2/carbon-identity-framework/pull/5318/commits/12e244d90280f7810c7e2c48fed01ce4eb48f957